pkg/virt-operator/resource/apply/delete_test.go

```diff
@@ -0,0 +1,185 @@
+						Finalizers: []string{v1.VirtOperatorComponentFinalizer},
+					},
+				},
+			}
+
+			needAdded := crdFilterNeedFinalizerAdded(crds)
+
+			Expect(len(needAdded)).To(Equal(1))
+			Expect(needAdded[0].Name).To(Equal("needs-finalizer"))
+		})
+
+		It("Filter needs finalizer removed", func() {
+			crds := []*extv1.CustomResourceDefinition{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "1",
+						Finalizers:        []string{v1.VirtOperatorComponentFinalizer},
+						DeletionTimestamp: now(),
+					},
+					Status: extv1.CustomResourceDefinitionStatus{
+						Conditions: []extv1.CustomResourceDefinitionCondition{
+							instanceRemovedCondition(),
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "2",
+						Finalizers:        []string{v1.VirtOperatorComponentFinalizer},
+						DeletionTimestamp: now(),
+					},
+					Status: extv1.CustomResourceDefinitionStatus{
+
+						Conditions: []extv1.CustomResourceDefinitionCondition{
+							instanceRemovedCondition(),
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "3",
+						Finalizers:        []string{v1.VirtOperatorComponentFinalizer},
+						DeletionTimestamp: now(),
+					},
+					Status: extv1.CustomResourceDefinitionStatus{
+						Conditions: []extv1.CustomResourceDefinitionCondition{
+							instanceRemovedCondition(),
+						},
+					},
+				},
+			}
+
+			needRemoved := crdFilterNeedFinalizerRemoved(crds)
+			Expect(len(needRemoved)).To(Equal(3))
+		})
+
+		It("Should block finalizer removal until all CRD CRs are removed", func() {
+			crds := []*extv1.CustomResourceDefinition{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "1",
+						Finalizers:        []string{v1.VirtOperatorComponentFinalizer},
+						DeletionTimestamp: now(),
+					},
+					Status: extv1.CustomResourceDefinitionStatus{
+						Conditions: []extv1.CustomResourceDefinitionCondition{
+							instanceRemovedCondition(),
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "2",
+						Finalizers:        []string{v1.VirtOperatorComponentFinalizer},
+						DeletionTimestamp: now(),
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "3",
+						Finalizers:        []string{v1.VirtOperatorComponentFinalizer},
+						DeletionTimestamp: now(),
+					},
+					Status: extv1.CustomResourceDefinitionStatus{
+						Conditions: []extv1.CustomResourceDefinitionCondition{
+							instanceRemovedCondition(),
+						},
+					},
+				},
+			}
+
+			needRemoved := crdFilterNeedFinalizerRemoved(crds)
+			Expect(len(needRemoved)).To(Equal(0))
+		})
+	})
+})
+
+func instanceRemovedCondition() extv1.CustomResourceDefinitionCondition {
+	return extv1.CustomResourceDefinitionCondition{
+		Type:   extv1.Terminating,
+		Status: extv1.ConditionFalse,
+		Reason: "InstanceDeletionCompleted",
+	}
+}
+
+func now() *metav1.Time {
```

Comment:
This is here in order to perform `now()` instead of `&metav1.Now()`?

Reply:
yes. otherwise the code won't compile. example

```pkg/virt-operator/resource/apply/delete_test.go:158:26: cannot take the address of "kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/apis/meta/v1".Now()
```